### PR TITLE
Parameterized Tests

### PR DESCRIPTION
**Pull request by Sumukh-Phalgaonkar**
Parameterized PostgresSinkConsumerIT to run with both CDCSDK server and Kafka Connect

### CODE REVIEW

File: src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java [hunk 1]
```diff
@@ -96,9 +90,40 @@ public static void afterClass() throws Exception {
         ybHelper.execute(UtilStrings.getDropTableStmt(DEFAULT_TABLE_NAME));
     }
 
-    @Test
+    @ParameterizedTest
+    @ValueSource(strings = {"CDCSDK" ,"Kafka connect"})
```
Comment (fourpointfour) [inline src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java]: Instead of passing strings directly, use ENUMs. Maybe create a java file just for `RunnerParameter.java` and then use value sources here like `RunnerParameter.CDCSDK, RunnerParameter.Connect`

Comment (fourpointfour) [reply src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java]: This way you will not need to worry about the values if someone even modifies it accidentally.

File: src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java [hunk 2]
```diff
@@ -160,4 +184,16 @@ public void verifyRecordsInPostgresFromKafka() throws Exception {
...
+
+    public void getExpectedDataInKafka() {
+        for (int i = 0; i < recordsToBeInserted; i++) {
+            Map<String, Object> expectedRecord = new LinkedHashMap<String, Object>();
+            expectedRecord.put("id", (i + recordNumber));
+            expectedRecord.put("first_name", new String("first_" + (i + recordNumber)));
+            expectedRecord.put("last_name", new String("last_" + (i + recordNumber)));
+            expectedRecord.put("days_worked", Double.valueOf(23.45));
+            expectedDataInKafka.add(expectedRecord);
+
```
Comment (fourpointfour) [inline src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java]: Remove extra line

File: src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java [hunk 3]
```diff
@@ -96,9 +90,40 @@ public static void afterClass() throws Exception {
...
+            cdcsdkContainer.withNetwork(containerNetwork);
+            cdcsdkContainer.start();
+            recordNumber = 0;
+            getExpectedDataInKafka();
+        }
+
+        //  Wait for bootstrapping to take place
+         Awaitility.await().atMost(Duration.ofSeconds(30)).pollDelay(Duration.ofSeconds(20)).until(()->true);
+
+
```
Comment (fourpointfour) [inline src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java:111]: Extra line

File: src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java [hunk 4]
```diff
@@ -96,9 +90,40 @@ public static void afterClass() throws Exception {
...
+        // Assuming that yugabyted is running.
+        ybHelper.execute(UtilStrings.getCreateTableYBStmt(DEFAULT_TABLE_NAME));
+
+        if (param.equals("Kafka connect")) {
+            // Set source connector config.
+            sourceConnector = kafkaHelper.getSourceConfiguration(ybHelper, "public." + DEFAULT_TABLE_NAME);
+            kafkaConnectContainer.registerConnector("source-connector", sourceConnector);
+            recordNumber = 5;
+            getExpectedDataInKafka();
+
```
Comment (fourpointfour) [inline src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java]: Extra line

File: src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java [hunk 5]
```diff
@@ -35,15 +40,20 @@
  * Release test that verifies basic reading from a YugabyteDB database and
  * writing to Kafka and then further to a PostgreSQL sink database
  *
- * @author Isha Amoncar, Vaibhav Kushwaha (vkushwaha@yugabyte.com)
+ * @author Isha Amoncar, Sumukh Phalgaonkar, Vaibhav Kushwaha (vkushwaha@yugabyte.com)
  */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
Comment (fourpointfour) [inline src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java:49]: Why is this required? I thought JUnit was able to understand the simple `@Order` annotation without specifying any orderer class.

**Review by fourpointfour — CHANGES_REQUESTED** [review]

File: src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java (hunk 5 above)
Comment (Sumukh-Phalgaonkar) [reply src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java:49]: No `@Order` was not working without this. Earlier inserts where happening in `@BeforeAll` block so the order of running tests didn't matter.  I moved the inserts inside first test hence order of execution became important.

File: src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java (hunk 1 above)
Comment (Sumukh-Phalgaonkar) [reply src/test/java/com/yugabyte/cdcsdk/testing/PostgresSinkConsumerIT.java]: I have moved this strings to UtilStrings class and now using them as value source

**Comment by vrajat** [comment]
The current change looks much more manageable to me. What does the test o/p look like?

**Comment by Sumukh-Phalgaonkar** [comment]
> The current change looks much more manageable to me. What does the test o/p look like?

Same as last time
```[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.yugabyte.cdcsdk.testing.PostgresSinkConsumerIT
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/ec2-user/.m2/repository/org/jboss/slf4j/slf4j-jboss-logmanager/1.1.0.Final/slf4j-jboss-logmanager-1.1.0.Final.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/ec2-user/.m2/repository/org/slf4j/slf4j-reload4j/1.7.36/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Slf4jLoggerFactory]
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 429.508 s - in com.yugabyte.cdcsdk.testing.PostgresSinkConsumerIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  07:26 min
[INFO] Finished at: 2022-12-19T07:42:30Z
[INFO] ------------------------------------------------------------------------```

**Comment by vrajat** [comment]
Is there a way to find out the name of the tests?

**Comment by Sumukh-Phalgaonkar** [comment]
> Is there a way to find out the name of the tests?

No, but when it fails, it shows the name of test where it failed

**Comment by vrajat** [comment]
Can you add documentation for the following in the README ?
* How to make a test run for both, CDCSDK only, Kafka Connect only
* How to run a single instance of the test.

**Review by vrajat — APPROVED** [review]
